modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/AbstractSlotManager.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.andes.kernel.slot;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.andes.configuration.AndesConfigurationManager;
+import org.wso2.andes.configuration.enums.AndesConfiguration;
+import org.wso2.andes.kernel.AndesException;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * AbstractSlotManager holds the common responsibilities for the standalone slot manager and cluster mode slot manager.
+ */
+public abstract class AbstractSlotManager {
+
+    /**
+     * To keep the name of queue on which deletion task is currently running
+     */
+    protected String currentDeletionQueue = StringUtils.EMPTY;
+
+    /**
+     * To keep the lower bound Id of the current expiry message deletion range
+     */
+    protected long currentDeletionRangeLowerBoundId;
+
+    /**
+     *  The slot gap need to be maintained as not reachable for deletion task as they may be allocated
+     *  to a slot delivery worker in the near future
+     */
+    int safetySlotCount = AndesConfigurationManager.readValue
```

Comment:
even if both approaches will do the same thing, I would rather 'init' this variable in constructor ( like we have done in other areas in the code)